### PR TITLE
[BugFix]choose alive node in default warehouse

### PR DESCRIPTION
**Pull request by abc982627271**
Fixes #issue

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 29 (13.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarRoutineLoadJob.java|0|5|00.00%|[215, 216, 217, 218, 220]|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|4|00.00%|[187, 188, 189, 191]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|0|5|00.00%|[235, 236, 237, 238, 240]|
|:large_blue_circle:|com/starrocks/common/util/PulsarUtil.java|0|4|00.00%|[145, 146, 147, 149]|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|1|5|20.00%|[104, 105, 106, 108]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|3|6|50.00%|[179, 180, 182]|

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/http/rest/LoadAction.java [hunk 1]
```diff
@@ -97,10 +99,15 @@ public void executeWithoutPassword(BaseRequest request, BaseResponse response) t
...
+        List<Long> nodeIds = new ArrayList<>();
         if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
             Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
-            nodeIds = warehouse.getAnyAvailableCluster().getComputeNodeIds();
+            for (long nodeId : warehouse.getAnyAvailableCluster().getComputeNodeIds()) {
+                if (GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNode(nodeId).isAvailable()) {
+                    nodeIds.add(nodeId);
+                }
+            }
+            Collections.shuffle(nodeIds);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/LoadAction.java:110]: move 113-115 code out from `else` branch.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 32 (18.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarRoutineLoadJob.java|0|5|00.00%|[215, 216, 217, 218, 220]|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|4|00.00%|[187, 188, 189, 191]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|0|5|00.00%|[235, 236, 237, 238, 240]|
|:large_blue_circle:|com/starrocks/common/util/PulsarUtil.java|0|4|00.00%|[145, 146, 147, 149]|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|3|8|37.50%|[105, 106, 107, 109, 110]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|3|6|50.00%|[179, 180, 182]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 32 (18.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarRoutineLoadJob.java|0|5|00.00%|[215, 216, 217, 218, 220]|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|4|00.00%|[187, 188, 189, 191]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|0|5|00.00%|[235, 236, 237, 238, 240]|
|:large_blue_circle:|com/starrocks/common/util/PulsarUtil.java|0|4|00.00%|[145, 146, 147, 149]|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|3|8|37.50%|[105, 106, 107, 109, 110]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|3|6|50.00%|[179, 180, 182]|

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java [hunk 2]
```diff
@@ -184,7 +184,11 @@ private PProxyResult sendProxyRequest(PProxyRequest request) throws UserExceptio
                 List<Long> nodeIds = new ArrayList<>();
                 if ((RunMode.getCurrentRunMode() == RunMode.SHARED_DATA)) {
                     Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
-                    nodeIds = warehouse.getAnyAvailableCluster().getComputeNodeIds();
+                    for (long nodeId : warehouse.getAnyAvailableCluster().getComputeNodeIds()) {
+                        if (GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNode(nodeId).isAlive()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:188]: should check `getBackendOrComputeNode` result is not null

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java [hunk 3]
```diff
@@ -232,7 +232,12 @@ public int calculateCurrentConcurrentTaskNum() throws MetaNotFoundException {
         int aliveNodeNum = systemInfoService.getAliveBackendNumber();
         if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
             Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
-            aliveNodeNum = warehouse.getAnyAvailableCluster().getComputeNodeIds().size();
+            aliveNodeNum = 0;
+            for (long nodeId : warehouse.getAnyAvailableCluster().getComputeNodeIds()) {
+                if (GlobalStateMgr.getCurrentSystemInfo().getBackendOrComputeNode(nodeId).isAlive()) {
+                    aliveNodeNum = aliveNodeNum + 1;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java:238]:
```suggestion
                    ++aliveNodeNum;
```

**Review by wyb — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 38 (15.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/PulsarRoutineLoadJob.java|0|6|00.00%|[216, 217, 218, 219, 220, 222]|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|5|00.00%|[187, 188, 189, 190, 192]|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|0|6|00.00%|[236, 237, 238, 239, 240, 242]|
|:large_blue_circle:|com/starrocks/common/util/PulsarUtil.java|0|5|00.00%|[145, 146, 147, 148, 150]|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|3|9|33.33%|[105, 106, 107, 108, 110, 111]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadManager.java|3|7|42.86%|[180, 181, 182, 184]|

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**